### PR TITLE
141 connections api to accept a connection

### PR DESCRIPTION
**Pull request by SnehaParuchuri**
Added **AcceptConnection** method, requestId and users validation is done and both **From** and **To** user's followers list is updated and the request is deleted from ConnectionRequests table.

@VishnuJammula please review.

### CODE REVIEW

File: backend/.vscode/settings.json [hunk 1]
```diff
@@ -11,6 +11,12 @@
             "driver": "SQLite",
             "name": "test1",
             "database": "${workspaceFolder:backend}/jobs.db"
+        },
```
Comment (VishnuJammula) [inline backend/.vscode/settings.json:14]: git ignore this file

File: backend/main.go [hunk 2]
```diff
@@ -61,5 +61,7 @@ func main() {
 	protected.POST("/applyjob", controllers.ApplyToJob)
 	protected.POST("/resumeupload/:id", controllers.UploadResume)
 
+	protected.POST("/requestconnection", controllers.RequestConnection)
+	protected.DELETE("/acceptconnection", controllers.AcceptConnection)
```
Comment (VishnuJammula) [inline backend/main.go:65]: Why is this api using DELETE verb? I think it should be POST

File: backend/controllers/connections-controller.go [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+}
+func followingCheck(followers pq.Int64Array, requestor int) bool {
+
+	for _, a := range followers {
+		fmt.Printf("%d   %d", a, requestor)
+		if int(a) == requestor {
+			return true
+		}
+	}
+	fmt.Print("here")
```
Comment (VishnuJammula) [inline backend/controllers/connections-controller.go:64]: Remove print statements

File: backend/controllers/connections-controller.go [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+	"github.com/gin-gonic/gin"
+	"github.com/lib/pq"
+)
+
+func RequestConnection(c *gin.Context) {
+	var request models.ConnectionRequest
+	c.BindJSON(&request)
+	// requestedFrom and requestedTo cannot be the same person
+	if request.RequestedFrom == request.RequestedTo {
+		c.JSON(400, gin.H{"error": "Requested From and To cannot be the same user!"})
```
Comment (VishnuJammula) [inline backend/controllers/connections-controller.go:19]: Can you replace 400 by http.{appropriate variable}. All the http codes are present in http class. Please do this everywhere you used numeric values

File: backend/controllers/connections-controller.go [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+	result1 := utils.DB.Where("requested_from = ? and requested_to = ?", request.RequestedTo, request.RequestedFrom).First(&existingRequest1)
+	if result != nil && result.RowsAffected == 1 {
+		c.JSON(400, gin.H{"error": "You have already requested a connection from the user"})
+		return
+	}
+	if result1 != nil && result1.RowsAffected == 1 {
+		c.JSON(400, gin.H{"error": "The user has already requested a connection from you. Please approve it."})
+		return
+	}
+	if !isUserPresent(request.RequestedFrom) {
```
Comment (VishnuJammula) [inline backend/controllers/connections-controller.go:35]: Please move the user presence checks before checking for existing requests

File: backend/controllers/connections-controller.go [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+		return
+	}
+	if !isUserPresent(request.RequestedTo) {
+		c.JSON(400, gin.H{"error": "Unable to find user with ID " + strconv.Itoa(request.RequestedTo)})
+		return
+	}
+	//already connected user check
+	var requestingUser models.User
+	utils.DB.Where("user_id = ?", request.RequestedFrom).First(&requestingUser)
+	fmt.Print(requestingUser.Following)
```
Comment (VishnuJammula) [inline backend/controllers/connections-controller.go:46]: remove print statements

File: backend/controllers/connections-controller.go [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+	if u == nil || u.RowsAffected != 1 {
+		c.JSON(400, gin.H{"error": "Did not find any user with Id " + strconv.Itoa(request.RequestedTo)})
+		return
+	}
+	if request.RequestedTo != existingRequest.RequestedTo {
+		c.JSON(400, gin.H{"error": "Connection can only be accepted by use with id " + strconv.Itoa(request.RequestedTo)})
+		return
+	}
+	//delete the request if the requested user does not exist
+	if !isUserPresent(existingRequest.RequestedFrom) {
```
Comment (VishnuJammula) [inline backend/controllers/connections-controller.go:88]: move user presence checks before performing any other logic with user ids

**Review by VishnuJammula — CHANGES_REQUESTED** [review]

**Comment by SnehaParuchuri** [comment]
@VishnuJammula updated the changes in controllers

**Pull request merged**